### ISSUE TITLE
Slow gallery loading

### ISSUE THREAD

**Issue by newNadik** [Mindinventory/Lassi-Android#56]
Got a report from the user about really slow gallery loading, after some time photos appear on the screen
- used library version: 0.4.0
- code: 

```
private fun attachFile(type: MediaType) {
        mFragment?.context?.let {
            val lassIntent = Lassi(it)
                .with(LassiOption.CAMERA_AND_GALLERY)
                .setGridSize(3)
                .disableCrop()
            when (type) {
                MediaType.VIDEO -> {
                    lassIntent.setMediaType(MediaType.VIDEO)
                }
                MediaType.IMAGE -> {
                    lassIntent.setMediaType(MediaType.IMAGE)
                }
                else -> {}
            }
            lassIntent.setStatusBarColor(R.color.SecondaryTextColor)
                .setToolbarColor(R.color.SecondaryTextColor)
                .setToolbarResourceColor(R.color.TextColor)
                .setProgressBarColor(R.color.SecondaryTextColor)
                .setPlaceHolder(R.drawable.ic_image_placeholder)
                .setErrorDrawable(R.drawable.ic_image_placeholder)
                .setSelectionDrawable(R.drawable.ic_checked_media)
            val intent = lassIntent.build()
            receiveData?.launch(intent)
        }
    }

    private val receiveData =
        mFragment?.registerForActivityResult(ActivityResultContracts.StartActivityForResult()) {
            if (it.resultCode == Activity.RESULT_OK) {
                val selectedMedia =
                    it.data?.getSerializableExtra(KeyUtils.SELECTED_MEDIA) as ArrayList<MiMedia>
                val result = ArrayList<KIOAttachment>()
                selectedMedia.forEach { media ->
                    result.add(KIOAttachment(miMedia = media))
                }
                completeListener?.invoke(result)
            }
        }
```

- device: Samsun S21
- video of the issue

https://user-images.githubusercontent.com/16311133/162269908-a27df476-f0c6-45ac-b812-db64a21bf8c9.mp4

 Thank you

**Comment by sanjay-mi** [comment]
Hello @newNadik, 

Thank you for using the Lassi library. 

We have analysed your reported issue. Here loading indicator will display based on the size of data in your device. We have noticed that, If 1k images are there in the device then it will take around 1 second to display but if 7k images are there then it will take approx 14 seconds. So depending on the size of the data, it will show a loading indicator.

Our team is working on the optimisation of loading time.

**Comment by thezeeshan92** [comment]
Yes, I have faced the same issue.  My mobile has approx 14k images, this library takes approx 50 sec to load these images, but on the other hand, I have used some other applications like WhatsApp, telegram, etc.  These applications load all images within 10 sec.

**Comment by sanjay-mi** [comment]
Hello @newNadik, @thezeeshan92 

We have released a new version of lassi 1.0.0. 
In this version, we have resolved this loading issue. The first time it will load all the files and from next time it will display them directly without a loading indicator.

Please try it out and let us know in case have any issues.

Thank you
